File: server/src/main/java/cn/iinti/atom/entity/ServerNode.java
package cn.iinti.atom.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * <p>
 * 服务器节点，多台服务器组成代理集群
 * </p>
 *
 * @author iinti
 * @since 2022-12-14
 */
@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
@TableName("server_node")
@Schema(name = "ServerNode对象", description = "服务器节点，多台服务器组成代理集群")
public class ServerNode implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    @Schema(name = "自增主建")
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @Schema(name = "服务器id，唯一标记服务器")
    private String serverId;

    @Schema(name = "最后心跳时间")
    private LocalDateTime lastActiveTime;

    @Schema(name = "创建时间")
    private LocalDateTime createTime;

    @Schema(name = "工作IP")
    private String ip;

    @Schema(name = "springboot 服务开启端口")
    private Integer port;

    @Schema(name ="服务器是否启用")
    private Boolean enable;

    @Schema(name = "本地ip")
    private String localIp;

    @Schema(name = "工作ip")
    private String outIp;


    public static final String ID = "id";

    public static final String SERVER_ID = "server_id";

    public static final String LAST_ACTIVE_TIME = "last_active_time";

    public static final String CREATE_TIME = "create_time";


    public static final String IP = "ip";

    public static final String PORT = "port";
    public static final String ENABLE = "enable";

    public static final String LOCAL_IP = "local_ip";

    public static final String OUT_IP = "out_ip";
}

File: server/src/main/kotlin/cn/iinti/atom/entity/ServerNode.kt
package cn.iinti.atom.entity

import com.baomidou.mybatisplus.annotation.IdType
import com.baomidou.mybatisplus.annotation.TableId
import com.baomidou.mybatisplus.annotation.TableName
import io.swagger.v3.oas.annotations.media.Schema
import lombok.Data
import lombok.EqualsAndHashCode
import lombok.experimental.Accessors
import java.io.Serial
import java.io.Serializable
import java.time.LocalDateTime

@Schema(name = "ServerNode对象", description = "服务器节点，多台服务器组成代理集群")
@TableName("server_node")
@Accessors(chain = true)
@Data
@EqualsAndHashCode(callSuper = false)
data class ServerNode(
    @Schema(name = "自增主建")
    @TableId(value = "id", type = IdType.AUTO)
    var id: Long? = null,

    @Schema(name = "服务器id，唯一标记服务器")
    var serverId: String? = null,

    @Schema(name = "最后心跳时间")
    var lastActiveTime: LocalDateTime? = null,

    @Schema(name = "创建时间")
    var createTime: LocalDateTime? = null,

    @Schema(name = "工作IP")
    var ip: String? = null,

    @Schema(name = "springboot 服务开启端口")
    var port: Int? = null,

    @Schema(name = "服务器是否启用")
    var enable: Boolean? = null,

    @Schema(name = "本地ip")
    var localIp: String? = null,

    @Schema(name = "工作ip")
    var outIp: String? = null
) : Serializable {

    companion object {
        const val ID = "id"
        const val SERVER_ID = "server_id"
        const val LAST_ACTIVE_TIME = "last_active_time"
        const val CREATE_TIME = "create_time"
        const val IP = "ip"
        const val PORT = "port"
        const val ENABLE = "enable"
        const val LOCAL_IP = "local_ip"
        const val OUT_IP = "out_ip"
    }
}